sequencer/src/lib.rs

```diff
@@ -286,53 +293,64 @@ pub async fn init_node(
     let public_ip = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
     let orchestrator_client = OrchestratorClient::new(validator_args, public_ip.to_string());
 
-    let (mut config, wait_for_orchestrator) = match persistence.load_config().await? {
+    let private_staking_key = network_params.private_staking_key;
+    let public_staking_key = BLSPubKey::from_private(&private_staking_key);
+
+    let (config, wait_for_orchestrator) = match persistence.load_config().await? {
         Some(config) => {
             tracing::info!("loaded network config from storage, rejoining existing network");
             (config, false)
         }
         None => {
             tracing::info!("loading network config from orchestrator");
-            let config = orchestrator_client.get_config(public_ip.to_string()).await;
+            let mut config: NetworkConfig<VerKey, StaticElectionConfig> =
+                orchestrator_client.get_config(public_ip.to_string()).await;
+
+            // Get updated config from orchestrator containing all peer's public keys
+            config = orchestrator_client
+                .post_and_wait_all_public_keys(config.node_index, public_staking_key)
+                .await;
+
+            config.config.my_own_validator_config.private_key = private_staking_key.clone();
+            config.config.my_own_validator_config.public_key = public_staking_key;
+
```

Comment:
And also here, I don't think `config.config.my_own_validator_config` need to be assigned again if we're getting config from orchestrator.

Reply:
Since they are the same this shouldn't break anything though.

Reply:
At least the private key does need to be assigned right? The orchestrator wouldn't be able to set that itself